src/components/Authentication/Authentication.vue

```diff
@@ -0,0 +1,69 @@
+<template lang="pug">
+.girder-authentication-component
+  v-card
+    v-tabs(v-model="activeTab", color="primary", slider-color="yellow", dark)
+      v-tab(key="login") Log In
+      v-tab(key="registration", v-if="register") Register
+      v-tab-item(key="login-box")
+        login(ref="loginForm",
+            :oauth-providers="oauthProviders",
+            :forgot-password-url="forgotPasswordUrl",
+            :forgot-password-route="forgotPasswordRoute",
+            @forgotpassword="$emit('forgotpassword')")
+      v-tab-item(key="registration-box", v-if="register")
+        registration(ref="registerForm",
+            :oauth-providers="oauthProviders")
+</template>
+
+<script>
+import Login from './Login.vue';
+import Registration from './Register.vue';
```

Comment:
These are internal details, but for the sake of propagating best practices, we should use:

```javascript
import GirderLogin from './Login.vue';
import GirderRegistration from './Register.vue';

...
components: {
  GirderLogin,
  GirderRegistration,
},
```

And same for the OAuth component usage.

Thanks @jeffbaumes for teaching me this practice, which is [recommended by Vue](https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential).